packages/analytics-js/src/app/RudderAnalytics.ts

```diff
@@ -181,16 +182,20 @@ class RudderAnalytics implements IRudderAnalytics<IAnalytics> {
       options: autoTrackOptions = {},
       pageLifecycle,
     } = autoTrack ?? {};
+
     const {
       events = [PageLifecycleEvents.LOADED, PageLifecycleEvents.UNLOADED],
-      enabled = autoTrackEnabled,
+      enabled: pageLifecycleEnabled = autoTrackEnabled,
       options = autoTrackOptions,
     } = pageLifecycle ?? {};
 
-    state.autoTrack.enabled.value = autoTrackEnabled;
-    state.autoTrack.pageLifecycle.enabled.value = enabled;
+    // Set the autoTrack enabled state
+    // if at least one of the autoTrack options is enabled
+    state.autoTrack.enabled.value = autoTrackEnabled || pageLifecycleEnabled;
```

Comment:
We shouldn't do this. In the future when we will have more tracking options here, those will be enabled unknowingly.

Reply:
If the page life cycle is automatically tracking, it means auto tracking is enabled in the SDK, in some or the other form.

In the future, when more components are auto-tracked, they all have to be processed before setting this flag.
Anyway, in that scenario, we have to handle the whole functionality more elegantly.

Should I add a note in the code to avoid any risk?

Reply:
Yes, let's add a note